examples/Components/Typeahead/Typeahead.purs

```diff
@@ -2,178 +2,198 @@ module Docs.Components.Typeahead where
+            void $ H.query _select unit $ H.tell $ Select.PerformAction $ Select.setVisibility Select.Off
 
-          if length (filter ((==) item) st.items) > 0
-            then H.modify_ _ { selected = ( item : st.selected ) }
-            else H.modify_ _
-                  { items = ( item : st.items )
-                  , selected = ( item : st.selected ) }
+          case length (filter ((==) item) st.items) of
+            0 ->
+              H.modify_ _ { items = ( item : st.items ), selected = ( item : st.selected ) }
+            _ ->
+              H.modify_ _ { selected = ( item : st.selected ) }
 
           newSt <- H.get
           let newItems = difference newSt.items newSt.selected
-          _ <- H.query unit $ Select.replaceItems newItems
-          pure unit
-
-        otherwise -> pure unit
+          void $ H.query _select unit $ H.tell $ Select.ReplaceItems newItems
+      
+      , visibilityChanged: \_ -> pure unit
+      } msg
 
-      Removed item a -> do
-        st <- H.get
-        H.modify_ _ { selected = filter ((/=) item) st.selected }
-        newSt <- H.get
-        let newItems = difference newSt.items newSt.selected
-        _ <- H.query unit $ Select.replaceItems newItems
-        pure a
+    Remove item -> do
+      newSt <- H.modify \oldSt -> oldSt { selected = filter ((/=) item) oldSt.selected }
+      void $ H.query _select unit $ H.tell $ Select.ReplaceItems $ difference newSt.items newSt.selected
 
 
 ----------
 -- Helpers
 
-class_ :: ∀ p i. String -> H.IProp ( "class" :: String | i ) p
+class_ :: ∀ p i. String -> HH.IProp ( "class" :: String | i ) p
 class_ = HP.class_ <<< HH.ClassName
 
 filterItems :: TypeaheadItem -> Array TypeaheadItem -> Array TypeaheadItem
 filterItems str = filter (\i -> contains (Pattern str) i)
 
-renderInputContainer :: Select.State TypeaheadItem -> Select.ComponentHTML Query TypeaheadItem
-renderInputContainer state = HH.div_ [ renderInput, renderContainer ]
+renderSelect 
+  :: forall m
+   . Select.State TypeaheadItem 
+  -> H.ComponentHTML (Select.Action TypeaheadItem ExtraActions () m) () m
+renderSelect state = HH.div_ [ renderInput, renderContainer ]
   where
     renderInput = HH.input $ Setters.setInputProps
       [ HP.classes CSS.input
-      , HP.placeholder "Type to search..." ]
+      , HP.placeholder "Type to search..." 
+      ]
 
     renderContainer =
-      HH.div [ class_ "relative z-50" ]
-      $ if state.visibility == Select.Off
-        then []
-        else [ renderItems $ renderItem `mapWithIndex` state.items ]
+      HH.div 
+        [ class_ "relative z-50" ]
+        if state.visibility == Select.Off
+          then []
+          else [ renderItems $ renderItem `mapWithIndex` state.items ]
       where
-        renderChild =
-          HH.div
-          [ HE.onClick $ Select.always $ Select.raise $ H.action $ Log "I was clicked" ]
+      renderChild =
+        HH.div
+          [ HE.onClick \_ -> Just $ log "I was clicked" ]
```

Comment:
The newly-extended action can be used just like any other Select action now. (You can also recursively call other Select actions in the handler).